Classes/ViewHelpers/Resource/LanguageViewHelper.php

```
@@ -0,0 +1,199 @@
+		$this->registerArgument('path', 'string', 'Absolute or relative path to the locallang file', FALSE, self::LOCALLANG_DEFAULT);
+		$this->registerArgument('languageKey', 'string', 'Key for getting translation of a different than current initialized language', FALSE, NULL);
+		$this->registerArgument('as', 'string', 'If specified, a template variable with this name containing the requested data will be inserted instead of returning it.', FALSE, NULL);
+	}
+
+	/**
+	 * The main render method of this ViewHelper.
+	 *
+	 * @return array
+	 */
+	public function render() {
+		$as = $this->arguments['as'];
+		$path = $this->resolvePath($this->arguments['path']);
+
+		$languageKey = $this->getLanguageKey();
+		$locallang = GeneralUtility::readLLfile($path, $languageKey);
+		$labels = $this->getLabelsByLanguageKey($locallang, $languageKey);
+		$labels = $this->getLabelsFromTarget($labels);
+
+		if (FALSE === empty($as)) {
+			$labels = ViewHelperUtility::renderChildrenWithVariables($this, $this->templateVariableContainer, array($as => $labels));
+		}
+
+		return $labels;
+	}
+
+	/**
+	 * Resolves the extension name from defined argument or
+	 * tries to get the name from controller context if not set.
+	 *
+	 * @param string $extensionName
+	 * @return string
+	 * @throws Exception
+	 */
+	protected function resolveExtensionName($extensionName) {
+		if (NULL === $extensionName) {
+			$request = $this->controllerContext->getRequest();
+			$extensionName = $request->getControllerExtensionName();
+		}
+
+		if (TRUE === empty($extensionName)) {
+			throw new Exception('Unable to read extension name from ControllerContext and value not manually specified');
+		}
+
+		return $extensionName;
+	}
+
+	/**
+	 * Tries to resolve the file path if file does not exists.
+	 *
+	 * @param string $path
+	 * @return string
+	 */
+	protected function resolvePath($path) {
+		if (FALSE === @is_file(GeneralUtility::getFileAbsFileName($path))) {
```

Silencing errors is discouraged